tracer/src/Datadog.Trace/TraceProcessors/Obfuscator.cs

```diff
@@ -0,0 +1,190 @@
+// <copyright file="Obfuscator.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections;
+using System.Collections.Specialized;
+using System.Text;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.TraceProcessors
+{
+    // https://github.com/DataDog/dd-trace-java/blob/35487fa08f16503105b2ff37fb084ffa5c894f24/internal-api/src/main/java/datadog/trace/api/normalize/SQLNormalizer.java
+
+    internal class Obfuscator
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<Obfuscator>();
+        private static readonly UTF8Encoding Encoding = new UTF8Encoding(false);
+
+        private static BitArray numericLiteralPrefix = new BitArray(255, false);
+        private static BitArray splitters = new BitArray(255, false);
+
+        static Obfuscator()
+        {
+            char[] byteArray1 = new char[13] { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '-', '+', '.' };
+            char[] byteArray2 = new char[4] { ',', '(', ')', '|' };
+
+            for (int i = 0; i < byteArray1.Length; ++i)
+            {
+                int unsigned = byteArray1[i] & 0xFF;
+                numericLiteralPrefix.Set(unsigned, true);
+            }
+
+            for (int i = 0; i < byteArray2.Length; ++i)
+            {
+                int unsigned = byteArray2[i] & 0xFF;
+                splitters.Set(unsigned, true);
+            }
+
+            for (int i = 0; i < 256; ++i)
+            {
+                if (char.IsWhiteSpace((char)i))
+                {
+                    splitters.Set(i, true);
+                }
+            }
+        }
+
+        public static string SqlObfuscator(string sql)
+        {
+            var utf8 = Encoding.GetBytes(sql);
```

Comment:
The Java version of the code applies the filtering on the UTF8 bytes because it needs to return an `UTF8BytesString` (so it saves one encoding step).
Here you're returning a string, so encoding the string is just adding extra overhead. You should instead convert the string to a char array:
```
var str = sql.ToCharArray();
```
Then apply the same algorithm on the char array. Later we can even add extra logic to have that array allocated lazily

Reply:
Hey @kevingosse thank you for the comments! In my lasted commit you can see the changes I've applied:
https://github.com/DataDog/dd-trace-dotnet/pull/2498/commits/c6fc4e4f8442702d02321f5263f441cff06fb8cc 

One thing to note is that for the `IsSplitter` function I end up adding a logic to return `false` if converting the char to bye/int leads to a value larger than the index of the array it is supposed to for the character in.

When I used to do the encoding to byte the Kanji characters there were converted to 3 different bytes `[231, 189, 191]` representing it, but now that we manipulate the Kanji character directly when converted to byte/int the result is `[32639]` and that would lead to a `IndexOutOfRangeException` for which I simply added the `if` to prevent  using this large index numbers, please let me know your thoughts on that.